LokiLabs/microsoft-graph-explorer-teams-app
# Teams App Localization

Localization now works in the BROWSER & TEAMS APP. It was more painful to set up than it sounds 🙃 BUT IT IS DONE..... AND IT WORKs.. AND LIFE IS HAPPIER NOW ヾ(•ω•`)o

Some images to show how it will look~

English (Any English country works!):
![image](https://user-images.githubusercontent.com/49354780/126711515-d155d78a-f3e1-4e4a-9ae7-15bb9688c2c0.png)

Chinese (Any Chinese country works!):
![image](https://user-images.githubusercontent.com/49354780/126711450-cf54e17e-a025-4a72-9c90-4b25dfc6fa34.png)

**To test it:**
- In the teams app, observe that it starts off in English (given that you are in Canada, after all).
- Go to your Teams settings and change the language. On this specific branch, the test file / language is Chinese, but you can test with any supported language if you add the localization for it in the respective file.
- Observe that the language has changed after the language changes

**Extra testing:**
- In the browser, test on the localhost page; This is likely going to be `https://localhost:3000/#/tab`, but it depends which port you are hosting on.
- You should notice that it defaults to the content being English
- Change the URL to a different localization; To do so, simply add the locale code before the #: `https://localhost:3000/zh-CN/#/tab`
- Observe that the language is now in Chinese (or whatever you changed it to)


**Final note:**
This localization set up requires the supported languages to be populated with the appropriate text in order for a smooth UX experience to occur. It does not default missing localization items to the English localization. If the language is labelled as supported (in `i18n.js`) and a user switches to that language without there being a fully populated language file, the unpopulated text fields will simply display as their code (such as `Table of Contents.Query Runner`).

~

Associated task:
[AD#38594](https://dev.azure.com/microsoftgarage/Intern%20GitHub/_sprints/taskboard/GI21%20-%20Graph%20Explorer/Intern%20GitHub/Y21-S/07?workitem=38594)
